### PR TITLE
Beef up Request's toString

### PR DESCRIPTION
**Pull request by JakeWharton**
Previously it unconditionally showed tags even if empty and would not show headers.

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/Request.kt [hunk 1]
```diff
@@ -109,7 +109,29 @@ class Request internal constructor(
       level = DeprecationLevel.WARNING)
   fun cacheControl(): CacheControl = cacheControl
 
-  override fun toString(): String = "Request{method=$method, url=$url, tags=$tags}"
+  override fun toString() = buildString {
+    append("Request{method=")
+    append(method)
+    append(", url=")
+    append(url)
+    if (headers.size != 0) {
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/Request.kt]: Annoyed this doesn't have `isEmpty` / `isNotEmpty`. Probably coming soon from me. This should behave more like a collection!

**Comment by JakeWharton** [comment]
Will fix tomorrow

File: okhttp/src/main/java/okhttp3/Request.kt [hunk 2]
```diff
@@ -109,7 +109,29 @@ class Request internal constructor(
...
+      append(", headers=[")
+      headers.forEachIndexed { index, (name, value) ->
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/Request.kt]: should this be Headers’ toString() ?

Comment (JakeWharton) [reply okhttp/src/main/java/okhttp3/Request.kt]: Probably, with it's current toString changing to some other function

**Comment by swankjesse** [comment]
Some predictable failures:
```
java.lang.AssertionError: 
Expecting:
 "[0 ms] callStart: Request{method=GET, url=http://localhost:54263/}"
to match pattern:
 "\[\d+ ms] callStart: Request\{method=GET, url=http://localhost:54263/, tags=\{\}\}"
	at okhttp3.logging.HttpLoggingInterceptorTest$LogRecorder.assertLogMatch(HttpLoggingInterceptorTest.java:851)
	at okhttp3.logging.LoggingEventListenerTest$LogRecorder.assertLogMatch(LoggingEventListenerTest.java:228)
	at okhttp3.logging.LoggingEventListenerTest.get(LoggingEventListenerTest.java:76)
```

**Pull request merged**